src/Jackalope/Node.php

```diff
@@ -529,12 +530,19 @@ public function setProperty($name, $value, $type = PropertyType::UNDEFINED, $val
         }
 
         if ($validate) {
+            if (is_array($value)) {
+                foreach ($value as $key => $v) {
```

Comment:
yes i know. but i need to do this before validating as well. i won't say its not a mess, but i feel its not worth doing much more refactoring if the end result will still be a mess.

the whole constructor and setup and validation logic needs to be remade for 2.0 i think.